### PR TITLE
Added helper class to use data in HDF5 datasets

### PR DESCRIPTION
**Pull request by jfsantos**
This class enables one to use slices of datasets in an HDF5 file as matrices for training/testing algorithms in keras.

Usage example:

```
X_train = HDF5Matrix(datapath, 'features', train_start, train_start+n_training_examples, normalizer=normalize_data)
y_train = HDF5Matrix(datapath, 'targets', train_start, train_start+n_training_examples)
```

The arguments are:
`datapath`: path to HDF5 file
`dataset`: name of the dataset in the HDF5 from which to take the data
`start` and `end`: indices to slice the given dataset
`normalizer`: function to be applied to the data coming from the dataset on-the-fly. Can be used for centering/standardizing/scaling data, but cannot change the dimensions of the data in any way.

### CODE REVIEW

**Comment by fchollet** [comment]
Looks good. I'll move some other utils to io_utils.
